gitbutler-ui/src/lib/vbranches/types.ts

```diff
@@ -215,7 +215,7 @@ export class RemoteFile {
 	}
 
 	get filename(): string {
-		return this.path.replace(/^.*[\\/]/, '');
+		return this.path.replace(/^.*[/]/, '');
```

Comment:
Note that elsewhere in the code, to obtain the filename, the path is being split by "/", and the last item taken - this change is consistent with that - I haven't tested this on Windows.

Reply:
Looks like `/` is used as a separator on Windows also (at least the paths that are passed to this part of the code):

![image](https://github.com/gitbutlerapp/gitbutler/assets/639945/40c0c841-4601-4c62-89f1-234a8608c232)

Reply:
> As for the handling in the frontend, we should definitely be checking which platform we're running on and switching the regex based off that, not assuming. That's really the only 'proper' solution here since any character except for / and \0 is a valid pathname character on *nix.

@Qix- FYI :)

Reply:
> Thanks, makes sense. Would it be possible if we go one step further and add a Tauri function that returns the path separator for the platform, and incorporate that into the path splitting function?

I take it you are referring to the change I introduced in this line (if not, please correct me :smile: ):
![image](https://github.com/gitbutlerapp/gitbutler/assets/639945/7226a48f-6798-48e9-a36b-19b869d028c8)
As far as I can tell from testing on Windows, `\` is not used as a path separator in the UI. (If you check my screenshot from above, which was taken on a Windows VM, the file path that gets logged to the console is `Sub/Folder/With space/hello world.txt`)

If I were to add a patch to split by `\` on Windows and on `/` on Linux, it would actually break things. Windows would start considering whole file paths a single file name.

Also note that there are more places already in the frontend codebase that use `/` for path separator, which do not differentiate by OS:

- [here](https://github.com/gitbutlerapp/gitbutler/blob/d2375a43610ebfaf185106c81e3b56ef277cf700/gitbutler-ui/src/lib/components/ShareIssueModal.svelte#L48)
- [here](https://github.com/gitbutlerapp/gitbutler/blob/d2375a43610ebfaf185106c81e3b56ef277cf700/gitbutler-ui/src/lib/utils/paths.ts#L5)
- [here](https://github.com/gitbutlerapp/gitbutler/blob/d2375a43610ebfaf185106c81e3b56ef277cf700/gitbutler-ui/src/lib/vbranches/filetree.ts#L46)
- [here](https://github.com/gitbutlerapp/gitbutler/blob/d2375a43610ebfaf185106c81e3b56ef277cf700/gitbutler-ui/src/lib/components/FileCardHeader.svelte#L19)
- ...

Reply:
@Qix- (sorry, forgot to mention you)

Reply:
Right but they're not really correct; the app was written with only MacOS and Linux in mind, we're in the process of making it cross platform. We should be doing the right thing here and doing proper path parsing that is platform-aware, instead of standardizing on forward slashes.

The biggest reason why is that there might be paths coming from the frontend (user input, etc.) that we need to parse correctly regardless of the separator(s) supported. On windows it's `[\\/]` and on *nix it's `/`, I don't really see any other way around it than to do a platform check.

Reply:
> we're in the process of making it cross platform

Aaaah, right, I see now. I managed to run it on a Windows VM without WSL (I can't test with WSL because of virtualization), and it seemed to work pretty well. Occasionally I had to right-click and refresh, and it was  a bit sluggish (I think that may be related to the VM, though, it took forever to compile).

> that there might be paths coming from the frontend (user input, etc.)

Has anything like this already been implemented? I personally can't foresee having to input a path manually in a text box or something similar... (But then again, I don't know what the plan is for GitButler.)

> On windows it's` [\\/]` and on *nix it's `/`, I don't really see any other way around it than to do a platform check.

My recommendation in this case would be to stick to the `/` separator - this is what is already working on Windows. Otherwise, you might have to rewrite a whole load of stuff which would be unneeded. So, when (if?) anyone enters a path manually in an input field, convert it to *nix format before passing it to the backend.

Anyway, if the need to use different path formats based on environment were required, I wouldn't add it in this PR, it's better to keep PRs small and related if possible (I only added that small change because it was actually fixing a visualization issue, it didn't break anything on Windows, and has small impact - it was only being used in two components). It would be better to change all the path handling logic in a separate PR all at once. However, I'm not 100% sure if this will be needed, TBH.

Adding support for detecting the OS from the frontend is trivial - the `navigator.userAgent` string can be queried:

![image](https://github.com/gitbutlerapp/gitbutler/assets/639945/ea9398f9-ca35-4b31-bfe2-851d3c125da3)
![image](https://github.com/gitbutlerapp/gitbutler/assets/639945/1758f505-2fe7-4be9-b753-0da4e49baa84)

Please let me know if there is an issue anywhere which requires OS detection so I can have a look into it (I can test on Linux and Windows inside VM without WSL).

@Qix-

Reply:
> Yes, namely when setting up a new project.

That seems to be working fine in Windows and in Linux.

> I'm less inclined to say this really solves the issue

What issue?

> cmd.exe notably does not accept <...>

At any point in your code are you launching another process in `cmd.exe` and passing in paths?